dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/FutureEMBuilder.java

```diff
@@ -335,8 +335,13 @@ public EntityManagerBuilder createEMBuilder() {
                         metadata = (ComponentMetaData) provider.metadataIdSvc //
                                         .getMetaData(metadataIdentifier);
                     } catch (IllegalStateException x) {
-                        if (System.nanoTime() - start > TimeUnit.SECONDS.toNanos(30))
-                            throw x;
+                        try { //Check if this is an EJB in a Web Module
+                            metadata = (ComponentMetaData) provider.metadataIdSvc //
+                                            .getMetaData(getMetadataIdentifier(false));
```

Comment:
It is confusing that we have a parameter named `webModule` that is set to `false` here when we are checking for whether the EJB is in a web module.